src/AzureAuth.Test/ExceptionListToStringConverterTest.cs

```diff
@@ -0,0 +1,407 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.AzureAuth.Test
+{
+    using System;
+    using System.Collections.Generic;
+    using FluentAssertions;
+    using Microsoft.Identity.Client;
+    using NUnit.Framework;
+
+    public class ExceptionListToStringConverterTest
+    {
+        [Test]
+        public void EmptyExceptionList()
+        {
+            IEnumerable<Exception> exceptions = new List<Exception>();
+
+            // Act
+            string result = ExceptionListToStringConverter.Execute(exceptions);
+
+            // Assert
+            result.Should().BeNull();
+        }
+
+        [Test]
+        public void NullExceptionList()
+        {
+            IEnumerable<Exception> exceptions = null;
+
+            // Act
+            string result = ExceptionListToStringConverter.Execute(exceptions);
+
+            // Assert
+            result.Should().BeNull();
+        }
+
+        [Test]
+        public void ExceptionList_WithOneException()
+        {
+            IEnumerable<Exception> exceptions = new List<Exception>()
+            {
+                new Exception("This is an exception"),
+            };
+
+            // Act
+            string result = ExceptionListToStringConverter.Execute(exceptions);
+            string expectedResult = @"[" +
+                                        @"{""Message"":""This is an exception""," +
+                                        @"""InnerException"":null," +
+                                        @"""ExceptionType"":""System.Exception""," +
+                                        @"""AADErrorCode"":null}" +
+                                    @"]";

```

Comment:
All of your string literals can be simplified - when using the `@` string literal symbol, you can do multi-line strings right in the code, so you do not need every line to be it's own complete string. IDK if this gets optimized away by the compiler or not (the string concatenation), but it will be easier to read and edit if it's 1 big string literal.

Reply:
Taking this one step further, we could actually not do string comparisons for these tests at all, but do comparisons against deserialized objects to validate that our JSON is correct.

Reply:
Tl;Dr discussion offline - actually let's not do string comparisons for our assertions, let's assert that we do have deserializable JSON.